### PR TITLE
[optim] deprecate `LRScheduler.print_lr`

### LINKED ISSUE #99270: Remove lr_scheduler.print_lr
### 📚 The doc issue

The purpose of `print_lr` is unclear, and arguments are undocumented.
Also, the API is counter-intuitive. For example, in python API, we do not use `print('hello', verbose=False)` to not print something, and here the `verbose` is even a required argument. Instead, the user should put an if condition before calling the print function.

Related thread: https://discuss.pytorch.org/t/how-to-use-print-lr-in-the-lr-scheduler/132420
Source: https://github.com/pytorch/pytorch/blob/master/torch/optim/lr_scheduler.py#L113-L124

### Suggest a potential alternative/fix

I suggest removing that function since users can still simply get the current LR via `get_last_lr` and design the printing message on their own.

cc @vincentqb @jbschlosser @albanD @janeyx99

### PR DESCRIPTION
**Pull request by david20571015**
Fixes #99270

_______
# Deprecation Note

As of this PR, `LRScheduler.print_lr()` is deprecated. Please use `LRScheduler.get_last_lr()` to access the learning rate.

### CODE REVIEW

File: torch/optim/lr_scheduler.py [hunk 1]
```diff
@@ -155,7 +155,17 @@ def print_lr(
...
+        """Display the current learning rate.
+        .. deprecated:: 2.4
+            ``print_lr()`` is deprecated. Please use ``get_last_lr()`` to access the
+            learning rate.
+        """
+        warnings.warn(
+            "`LRScheduler.print_lr()` is being deprecated. To fetch the learning rate, "
+            "please use `get_last_lr()` instead. For more details, "
+            "see https://github.com/pytorch/pytorch/issues/99270.",
+            DeprecationWarning,
```
Comment (janeyx99) [inline torch/optim/lr_scheduler.py:167]:
```suggestion
            UserWarning,
```
I think we normally use UserWarning here.

**Review by janeyx99 — COMMENTED** [review]
cc @GdoongMathew as this was a part of your proposal

File: torch/optim/lr_scheduler.py (hunk 1 above)
Comment (janeyx99) [reply torch/optim/lr_scheduler.py:167]: Reason being CPython hides deprecationwarnings lol

Comment (david20571015) [reply torch/optim/lr_scheduler.py]: okay!

**Comment by janeyx99** [comment]
Could you paste what the doc preview for the function looks like once the link in the pytorchbot comment populates (the docs build needs to finish first)?

image
It would be the python docs link here!

**Comment by david20571015** [comment]
@janeyx99, could you trigger the build CI again, I fixed the failure.

**Comment by janeyx99** [comment]
done!

**Comment by david20571015** [comment]
> Could you paste what the doc preview for the function looks like once the link in the pytorchbot comment populates (the docs build needs to finish first)?
> 
> image
> It would be the python docs link here!

**Review by janeyx99 — APPROVED** [review]
